src/main/LibsodiumEncryptor.ts

```diff
@@ -33,31 +26,53 @@ export class LibsodiumEncryptor implements Encryptor {
     params: EncryptParams<T, K>
   ): EncryptResult<T, K> {
     const { item, fieldsToEncrypt, dataKey } = params
+    const plaintextFields = { ...item } as Omit<T, K>
     const nonce = randombytes_buf(crypto_secretbox_NONCEBYTES)
     const encryptionKey = this.deriveKey(dataKey, KeyType.ENCRYPTION)
 
-    const encryptedFields: {
-      [P in K]: Uint8Array
-    } = {} as ItemWithEncryptedFields<T, K>
-
+    const itemToEncrypt: itemToEncrypt<T, K> = {} as itemToEncrypt<T, K>
     fieldsToEncrypt.forEach((fieldName) => {
       const fieldValue = item[fieldName]
       if (fieldValue !== undefined && fieldValue !== null) {
-        encryptedFields[fieldName] = crypto_secretbox_easy(
-          this.toBuffer(fieldValue),
-          nonce,
-          encryptionKey
-        )
+        itemToEncrypt[fieldName] = fieldValue
+        delete (plaintextFields as any)[fieldName]
       }
     })
 
+    const encryptedFields = crypto_secretbox_easy(
+      serialize(itemToEncrypt),
+      nonce,
+      encryptionKey
+    )
```

Comment:
This is the nonce fix. Instead of reusing the nonce in the loop, we use it once on all the fields that need to be encrypted.